

syntax = "proto3";

package user;

// The user services definition.
service User {
  
  rpc GetUsers(GetUsersRequest) returns (GetUsersResponse) {}
  rpc GetUserById(GetUserByIdRequest) returns (GetUserByIdResponse) {}
  rpc CreateUser(CreateUserRequest) returns (CreateUserResponse) {}
  rpc UpdateUser(UpdateUserRequest) returns (UpdateUserResponse) {}
  rpc DeleteUser(DeleteUserRequest) returns (DeleteUserResponse) {}
}

message UserData {
    string id = 1;
    string name = 2;
    string email = 3;
    string password = 4;
}

message GetUsersRequest {

}

message GetUsersResponse {
    repeated UserData users = 1;
}

message GetUserByIdRequest {
    string id = 1;
}

message GetUserByIdResponse {
    UserData user = 1;
}

message CreateUserRequest {
    UserData user = 1;
}

message CreateUserResponse {
    UserData user = 1;
}

message UpdateUserRequest {
    UserData user = 1;
}

message UpdateUserResponse {
    UserData user = 1;
}

message DeleteUserRequest {
     string id = 1;
}

message DeleteUserResponse {
    UserData user = 1;
}



